src/bootstrap/src/core/config/config.rs

```diff
@@ -1771,8 +1771,36 @@ impl Config {
                 std_features: std_features_toml,
             } = rust;
 
-            config.download_rustc_commit =
-                config.download_ci_rustc_commit(download_rustc, config.llvm_assertions);
+            // Before something like [Enable debug assertions on alt
+            // builds](https://github.com/rust-lang/rust/pull/131077) lands, alt rustc builds do
+            // *not* have rustc debug assertions enabled. We should not download an CI alt rustc if
+            // we need rustc to have debug assertions (e.g. for crashes test suite).
+            //
+            // Note that `rust.debug = true` (currently) implies `rust.debug-assertions = true`!
+            //
+            // This relies also on the fact that global default for `download-rustc` will be `false`
+            // if it's not explicitly set.
+            let debug_assertions_requested = matches!(rustc_debug_assertions_toml, Some(true))
+                || (matches!(debug_toml, Some(true))
+                    && !matches!(rustc_debug_assertions_toml, Some(false)));
+
+            if debug_assertions_requested {
+                if let Some(ref opt) = download_rustc {
+                    if opt.is_string_or_true() {
+                        println!(
```

Comment:
nit: We're not very consistent about this within bootstrap, but IMO messages for user consumption should be `eprintln`'d so they don't get put into `grep` etc. This is mostly a preference thing and it's currently pretty 50/50 so up to you if you want to change this

Reply:
Fixed.